com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/Watchpoint.java

```diff
@@ -249,4 +262,14 @@ public void setCompiledLogpointExpression(Object compiledExpression) {
     public Object getCompiledLogpointExpression() {
         return null;
     }
+
+    @Override
+    public Object getCompiledExpression(long threadId) {
+        return compiledExpressions.get(threadId);
+    }
+
+    @Override
+    public void setCompiledExpression(long threadId, Object compiledExpression) {
+        compiledExpressions.put(threadId, compiledExpression);
```

Comment:
put/get in multiple thread is error prone.

Reply:
nice catch!